profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_dgt_connector/plugins/tmgmt_dgt_connector.plugin.inc

```diff
@@ -0,0 +1,284 @@
+        foreach ($results as $result) {
+          if ($job = tmgmt_job_load($result->tjid)) {
+            if ($job->hasPoetryTranslator() && $job->getState() == 1) {
+              // Check the poetry status.
+              $result_ps = db_select('poetry_status', 'ps')
+                ->fields('ps', array('status'))
+                ->condition('tjiid', $result->tjiid, '=')
+                ->execute()
+                ->fetchAssoc();
+              if (empty($result_ps) || (!empty($result_ps) && $result_ps['status'] == POETRY_STATUS_MESSAGE_PEN)) {
+                return FALSE;
+              }
+            }
+          }
+        }
+      }
+    }
+    // All languages are supported at the moment.
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getNotCanTranslateReason(TMGMTJob $job) {
+    return t('An active translation for this item already exists');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function requestTranslation(TMGMTJob $job) {
+
+    // Request translation from Poetry.
+    $response = $this->sendRequest($job);
+
+    // Handling response.
+    $response_xml = simplexml_load_string($response);
+
+    // If it is not an xml response, the SOAP request failed.
+    if (!$response_xml) {
+      _tmgmt_poetry_tmgmt_job_reject(
+        $job,
+        "Rejected",
+        "There was an error with the SOAP service.",
+        $response
+      );
+    }
+    else {
+      $request = $response_xml->request;
+
+      $statuses = $request->status;
+
+      // Check the statuses of the response
+      // in order to verify that everything is correct.
+      foreach ($statuses as $status) {
+        $message = (string) $status->statusMessage;
+        if ($message != 'OK') {
+          _tmgmt_poetry_tmgmt_job_reject(
+            $job,
+            "Rejected",
+            "There was an error with the Poetry request.",
+            $message
+          );
+        }
+      }
+
+      if (isset($request->demandeId)) {
+        // Insert or update the mapping information.
+        $demande_id = (array) $request->demandeId;
+        $items = $job->getItems();
+        $item = array_shift($items);
+
+        $result = db_select('poetry_map', 'pm')
+          ->fields('pm', array('annee', 'numero', 'version', 'partie'))
+          ->condition('entity_type', $item->item_type, '=')
+          ->condition('entity_id', $item->item_id, '=')
+          ->execute()
+          ->fetchAssoc();
+        if (!$result) {
+          db_insert('poetry_map')
+            ->fields(array(
+              'entity_type' => $item->item_type,
+              'entity_id' => $item->item_id,
+              'annee' => $demande_id['annee'],
+              'numero' => $demande_id['numero'],
+              'version' => $demande_id['version'],
+              'partie' => $demande_id['partie'],
+            ))
+            ->execute();
+        }
+        else {
+          // @FIXME Do not overwrite data. Might not be for the same revision.
```

Comment:
> Might not be for the same revision.

What is to fix in fact because in poetry_map neither tmgmt_job nor tmgmt_job_item, there is no revision id?

Reply:
No idea, sorry.

Again, this method is "cloned" from TMGMTPoetryTranslatorPluginController. This @todo was there and I didn't get into this piece of code to evaluate it and change it.